src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -323,16 +323,56 @@ private static Condition handleNotContainsAll(NotContainsAllCondition notContain
     return cast(field, String[].class).notContains(valueArray);
   }
 
+  private static boolean isJsonbArray(EntityType entityType, FieldCondition<?> fieldCondition) {
+    // Retrieve the field/column definition for filtering
+    Field field = getFieldForFiltering(fieldCondition, entityType);
+
+    // Assuming the field has an `isJsonbArray` method or property
+   return field.getIsJsonbArray();
+  }
+
   private static Condition handleContainsAny(ContainsAnyCondition containsAnyCondition, EntityType entityType, org.jooq.Field<Object> field) {
-    var valueList = containsAnyCondition
-      .value()
-      .stream()
-      .map(val -> valueField(val, containsAnyCondition, entityType))
-      .toArray(org.jooq.Field[]::new);
-    var valueArray = cast(array(valueList), String[].class);
-    return arrayOverlap(cast(field, String[].class), valueArray);
+    // Check if the field is a JSONB array
+    boolean isJsonbArray = isJsonbArray(entityType, containsAnyCondition);
+
+    if (isJsonbArray) {
+      // Handle as JSONB array
+      var jsonbConditions = containsAnyCondition
+        .value()
+        .stream()
+        .map(val -> {
+          var jsonbValue = valueField(val, containsAnyCondition, entityType);
+          return String.format("(%s @> '[\"%s\"]'::jsonb)", field, jsonbValue);
```

Comment:
Rather than do this with `String.format()`, is there a way to use SQL variable binding here to make it safer (I'm not sure what the type of `jsonbValue` is, so I have no idea what will happen when you drop it into a string like this)